src/query/storage/index.go

```diff
@@ -123,9 +123,11 @@ func FetchOptionsToAggregateOptions(
 // FetchQueryToM3Query converts an m3coordinator fetch query to an M3 query.
 func FetchQueryToM3Query(
 	fetchQuery *FetchQuery,
+	options *FetchOptions,
 ) (index.Query, error) {
-	matchers := fetchQuery.TagMatchers
-	// If no matchers provided, explicitly set this to an AllQuery
+	withOpts := fetchQuery.WithAppliedOptions(options)
```

Comment:
Perhaps just `fetchQuery = fetchQuery.WithAppliedOptions(options)` so that code below doesn't accidently use `fetchQuery.TagMatchers` or anything like that? (Also can imagine in the future other things mutate the fetch query?)